lib/server/index.js

```diff
@@ -223,6 +227,11 @@ Server.prototype = {
     var config = this.config;
     var test_page = config.get('test_page');
 
+    // If there is an array of test pages, we cycle through them as the default
+    if (Array.isArray(test_page)) {
+      test_page = test_page[this.testPageCount % test_page.length];
+    }
```

Comment:
it seems like we would actually want a queue of pending segments to run, and available test runners pop off that queue. This would decouple the number of segments from the actual parallelism, as i can imagine wanted to be able to adjust the parallelism based on several factors (type of machine, current debug task etc.)